new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java

```diff
@@ -1327,5 +1352,149 @@ public static boolean deleteClusterRoleBinding(String name) {
+   * @param containerName The container in the Pod where the command is run. If no container
+   *     name is provided than the first container in the Pod is used.
+   * @param redirectToStdout copy Process output to stdout
+   * @param command The command to run
+   * @return result of command execution
+   * @throws IOException if an I/O error occurs.
+   * @throws ApiException if Kubernetes client API call fails
+   * @throws InterruptedException if any thread has interrupted the current thread
+   */
+  public static ExecResult exec(V1Pod pod, String containerName, boolean redirectToStdout,
+      String... command)
+      throws IOException, ApiException, InterruptedException {
+
+    // Execute command using Kubernetes API
+    KubernetesExec kubernetesExec = createKubernetesExec(pod, containerName);
+    final Process proc = kubernetesExec.exec(command);
+
+    final CopyingOutputStream copyOut =
+        redirectToStdout ? new CopyingOutputStream(System.out) : new CopyingOutputStream(null);
+
+    // Start a thread to begin reading the output stream of the command
+    Thread out = null;
+    try {
+      out =
+          new Thread(
+              () -> {
+                try {
+                  ByteStreams.copy(proc.getInputStream(), copyOut);
+                } catch (IOException ex) {
+                  logger.warning("Exception reading from input stream.", ex);
+                }
+              });
+      out.start();
+
+      // wait for the Process, which represents the executing command, to terminate
+      proc.waitFor();
+
+      // wait for reading thread to finish any last remaining output
+      if (out != null) {
+        out.join();
+      }
+
+      // Read data from Process's stdout
+      String stdout = read(copyOut.getInputStream());
+
+      // Read from Process's stderr, if data available
+      String stderr = (proc.getErrorStream().available() != 0) ? read(proc.getErrorStream()) : null;
+
+      return new ExecResult(proc.exitValue(), stdout, stderr);
+    } finally {
+      if (proc != null) {
+        proc.destroy();
+      }
+    }
+  }
+
+  /**
+   * Create an object which can execute commands in Kubertenes container.
+   *
+   * @param pod The pod where the command is run
+   * @param containerName The container in the Pod where the command is run. If no container
+   *     name is provided than the first container in the Pod is used.
+   * @return object for executing a command in a container of the pod
+   */
+  public static KubernetesExec createKubernetesExec(V1Pod pod, String containerName) {
+    KubernetesExec kubernetesExec = EXEC_FACTORY.create(pod, containerName);
+    kubernetesExec.setStdin(stdin);
+    kubernetesExec.setTty(tty);
+    return kubernetesExec;
+  }
+
   //------------------------
+
+  /**
+   * KubernetesExec object for executing a command in the container of a pod.
+   */
+  public static class KubernetesExecImpl extends KubernetesExec {
```

Comment:
an inline class that implements an abstract class defined in a different file seems really cumbersome

Reply:
Simplified the implementation as your suggestion below.